applications/serial_lte_modem/src/slm_util.c

```diff
@@ -213,3 +189,31 @@ int util_string_get(const struct at_param_list *list, size_t index,
 
 	return -ENOMEM;
 }
+
+/**
+ * @brief use AT command to get IPv4 address
+ */
+void util_get_addr(char *addr4, char *addr6)
+{
+	int err;
+	char rsp[128];
+	size_t addr_len;
+
+	err = at_cmd_write("AT+CGPADDR", rsp, sizeof(rsp), NULL);
+	if (err) {
+		return;
+	}
+	/* parse +CGPADDR: <cid>,<PDP_addr_1>,<PDP_addr_2> */
+	err = at_parser_params_from_str(rsp, NULL, &at_param_list);
+	if (err) {
+		return;
+	}
+	if (addr4) {
+		addr_len = NET_IPV4_ADDR_LEN;
+		(void)util_string_get(&at_param_list, 2, addr4, &addr_len);
```

Comment:
I just wonder, is it guaranteed that the first parameter is always IPv4 address? Looking at the AT command reference, it could be either IPv4 or IPv6:
```
<PDP_addr_1>
    String. For IPv4 given as a dot-separated numeric (0–255) parameter. For IPv6 given
    as a colon-separated hexadecimal (0x0000–0xFFFF) parameter.

```

Reply:
Yes the return all depends on PDN type.
Now I removed address type assumption and added IPv4/IPv6 check instead.